libqtile/config.py

```diff
@@ -610,32 +610,26 @@ def __init__(self, title=None, wm_class=None, role=None, wm_type=None,
 
     def compare(self, client):
         for _type, rule in self._rules:
-            if _type == "net_wm_pid":
-                def match_func(value):
-                    return rule == value
-            else:
-                match_func = getattr(rule, 'match', None) or \
-                    getattr(rule, 'count')
-
-            if _type == 'title':
+            # get value
+            if _type == "title":
                 value = client.name
-            elif _type == 'wm_class':
-                value = None
-                _value = client.window.get_wm_class()
-                if _value and len(_value) > 1:
-                    value = _value[1]
-            elif _type == 'wm_instance_class':
-                value = client.window.get_wm_class()
-                if value:
-                    value = value[0]
-            elif _type == 'wm_type':
-                value = client.window.get_wm_type()
-            elif _type == 'net_wm_pid':
-                value = client.window.get_net_wm_pid()
-            else:
+            elif _type == "wm_instance_class":
+                value = client.window.get_wm_class() and client.window.get_wm_class()[0]
```

Comment:
Doesn't this just take the first character of "wm_instance_class"?
```
>>> a = "foo"
>>> a and a[0]
'f'
```

Reply:
But there is no such thing as "wm_instance_class", it's the first class from the list, isn't it ?

Reply:
Right, thanks. I didn't realize this had just been changed from above.